### PR TITLE
Switch to mypy baseline

### PR DESCRIPTION
**Pull request by wouterdb**
# Description

I've been using this for months now and it works really well. 

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [ ] Attached issue to pull request
- [ ] Changelog entry
- [ ] Type annotations are present
- [ ] Code is clear and sufficiently documented
- [ ] No (preventable) type errors (check using make mypy or make mypy-diff)
- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )
- [ ] If this PR fixes a race condition in the test suite, also push the fix to the relevant stable branche(s) (see [test-fixes](https://internal.inmanta.com/development/core/tasks/build-master.html#test-fixes) for more info)

### CODE REVIEW

**Review by jptrindade — APPROVED** [review]

File: requirements.dev.txt [hunk 1]
```diff
@@ -9,6 +9,7 @@ types-setuptools==70.1.0.20240627
 types-toml==0.10.8.20240310
 flake8-junit-report==2.1.0
 time-machine==2.14.1
+mypy-baseline
```
Comment (hlloreda) [inline requirements.dev.txt:12]: Shouldn't we set (freeze) the version?

**Review by hlloreda — CHANGES_REQUESTED** [review]

File: requirements.dev.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (wouterdb) [reply requirements.dev.txt:12]: Perhaps I should just exclude iso6 from the change to keep it clean?

[the remark this answers is not in the document]
Comment (arnaudsjs) [reply requirements.dev.txt:12]: I think that would be best

**Review by hlloreda — APPROVED** [review]

**Review by arnaudsjs — APPROVED** [review]

**Review by Hugo-Inmanta — APPROVED** [review]

**Review by sanderr — APPROVED** [review]
Nice!

**Pull request closed without merging**